### PR TITLE
pbs_habitat does not call postinstall if PBS_HOME is empty

### PR DESCRIPTION
**Pull request by ashwathraop**
#### Describe Bug or Feature

If PBS_HOME directory is empty and pbs_habitat is run, script does not invoke pbs_postinstall and fails during db creation.

#### Describe Your Change

As part PBS_HOME directory extstance check also see if the directory is empty or not. If empty call pbs_postinstall.

Manual Test Output:

[test_log.txt](https://github.com/PBSPro/pbspro/files/3449526/test_log.txt)

### CODE REVIEW

File: src/cmds/scripts/pbs_habitat.in [hunk 1]
```diff
@@ -514,7 +514,15 @@ if [ -z "$PBS_HOME" ]; then
 	echo "*** PBS_HOME is not set"
 	exit 1
 fi
+
+update_home=0
 if [ ! -d "$PBS_HOME" ]; then
+        update_home=1
+else
+	# Check if PBS_HOME is empty.
+        [ ! "$(/bin/ls -A $PBS_HOME)" ] && update_home=1
```
Comment (hirenvadalia) [inline src/cmds/scripts/pbs_habitat.in]: How about you put this one in above if condition? like `if [ ! -d "${PBS_HOME}" -o ! -z "$(/bin/ls -A ${PBS_HOME})" ]; then`

Comment (ashwathraop) [reply src/cmds/scripts/pbs_habitat.in]: I thought of it but then just the sake of readability I broke the code. If you insist on merging the conditions I can do that :)

Comment (hirenvadalia) [reply src/cmds/scripts/pbs_habitat.in]: yes please merge it, about readability I think merged if will look more readable then current one

Comment (ashwathraop) [reply src/cmds/scripts/pbs_habitat.in]: Done. Please check now.

**Comment by prakashcv13** [comment]
merged #1220

**Pull request merged**